minos/common/model/fields.py

```diff
@@ -66,53 +65,75 @@ def value(self, data: t.Any) -> t.NoReturn:
-                f"Something goes wrong check if the type of the passed value "
-                f"is fine: data:{type(data)} vs type requested: {type_field}"
-            )
+            if type_field in PYTHON_INMUTABLE_TYPES:
+                if data is None or data == MissingSentinel:
+                    raise MinosReqAttributeException("")
+                elif type_field is int and self._is_int(data):
+                    log.debug("the Value passed is an integer")
+                    self._value = int(data)
+                    return
+                elif type_field is bool and self._is_bool(data):
+                    log.debug("the Value passed is an integer")
+                    self._value = data
+                    return
+                elif type_field is str and self._is_string(data):
+                    log.debug("the Value passed is a string")
+                    self._value = data
+                    return
+            else:
+                origin_type = t.get_origin(type_field)
+                if origin_type is None:
+                    raise MinosMalformedAttributeException("")
+                if data is None or data == MissingSentinel:
+                    raise MinosReqAttributeException("")
+                elif origin_type is list:
+                    converted_data = self._is_list(data, t.get_args(type_field)[0], convert=True)
+                    if isinstance(converted_data, bool) and not converted_data:
+                        raise MinosTypeAttributeException(
+                            f"{type(data)} could not be converted into {t.get_args(type_field)[0]} type"
+                        )
+                    self._value = converted_data
+                    return
+
+                elif origin_type is dict:
+                    converted_data = self._is_dict(
+                        data, t.get_args(type_field)[0], t.get_args(type_field)[1], convert=True
+                    )
+                    if isinstance(converted_data, bool) and not converted_data:
+                        raise MinosTypeAttributeException(
+                            f"{type(data)} could not be converted into {type_field} key, value types"
+                        )
+                    self._value = converted_data
+                    return
+                elif origin_type is ModelRef:
+                    converted_data = self._is_model_ref(data, t.get_args(type_field)[0], convert=True)
+                    if isinstance(converted_data, bool) and not converted_data:
+                        raise MinosTypeAttributeException(
+                            f"{type(data)} could not be converted into {t.get_args(type_field)[0]} type"
+                        )
+                    self._value = converted_data
+                    return
+
+        raise MinosTypeAttributeException(
+            f"The '{type_field}' type does not match with the given data type: {type(data)}"
+        )
```

Comment:
Hi @andrea-mucci, I think this ugly conditional code must be refactored into a sequence of clean and independent functions, but I've a doubt about  to refactor in this pull request, or if it's a better option to refactor it while resolving the #29 issue. 🤔

Reply:
UPDATE: I decided to refactor the code in this pull request as I realized that is independent respect to #29